dialects/linux/dproc.c

```diff
@@ -1395,6 +1397,57 @@ process_id(idp, idpl, cmd, uid, pid, ppid, pgid, tid, tcmd)
 	return(0);
 }
 
+/* compare namespace of this lsof proces and the target process */
+
+int
+compare_namespace(ns)
+        char *ns;                /* namespace */
+{
+        char nspath_self[NS_PATH_LENGTH], nspath_target[NS_PATH_LENGTH];
+        char ns_self[NS_LENGTH], ns_target[NS_LENGTH];
+        ssize_t n_self, n_target;
+        int ret;
+
+        ret = snprintf(nspath_self, NS_PATH_LENGTH, "/proc/self/ns/%s", ns);
+        if (ret >= NS_PATH_LENGTH || ret <= 0)
+	    return -1;
+
+        n_self = readlink(nspath_self, ns_self, NS_LENGTH);
+        if (n_self < 0)
+            return -1;
+
+        ret = snprintf(nspath_target, NS_PATH_LENGTH, "/proc/%d/ns/%s", Lp->pid, ns);
+        if (ret >= NS_PATH_LENGTH || ret <= 0)
+	    return -1;
+
+        n_target = readlink(nspath_target, ns_target, NS_LENGTH);
+        if (n_target < 0)
+            return -1;
+
+        if (n_self != n_target)
+            return -1;
+
+        if (strncmp(ns_self, ns_target, n_self))
+            return -1;
+
+        return 0;
+}
+
+/* compare kernel release */
+
+int
+compare_kernel_release(release)
```

Comment:
Do you want to do this in runtime or build time?

A funcitn is for runtime.
A macro is for build-time.

`LINUXV` is defined in build-time.
`release` is always 38000. So we can evaluate a result at build-time.
Defining this with a macro, you can improve the performance a bit.

However, what we would like to know is a fact at runtime.
Consider, a linux distribution ships a lsof binary package.
The distributor has a build farm that uses linux-3.7.
The new version of the distribution includes linux-5.x.
The runtime where a lsof binary runs is linux-5.x.
However, the lsof is built on linux-3.7.

Reply:
Yes, it's my carelessness to mix build-time and runtime. This is a typical runtime case, use uname(2) instead.